[Attack Discovery][Scheduling] Attack Discovery scheduling rule management (#12003)

e40pud opened the pull request:
## Summary

Main ticket ([Internal link](https://github.com/elastic/security-team/issues/12003))

To allow users to schedule Attack Discovery generations, we will use either [Alerting Framework](https://www.elastic.co/guide/en/kibana/current/alerting-getting-started.html). These changes add functionality to manage new alerts type - Attack Discovery Schedule.

### Introduced endpoints

- **Create** AD scheduling rule route: `POST /internal/elastic_assistant/attack_discovery/schedules`
- **Read/Get** AD scheduling rule by id route: `GET /internal/elastic_assistant/attack_discovery/schedules/{id}`
- **Update** AD scheduling rule by id route: `PUT /internal/elastic_assistant/attack_discovery/schedules/{id}`
- **Delete** AD scheduling rule by id route: `DELETE /internal/elastic_assistant/attack_discovery/schedules/{id}`
- **Enable** AD scheduling rule by id route: `POST /internal/elastic_assistant/attack_discovery/schedules/{id}/_enable`
- **Disable** AD scheduling rule by id route: `POST /internal/elastic_assistant/attack_discovery/schedules/{id}/_disable`
- **Find** all existing AD scheduling rules route: `GET /internal/elastic_assistant/attack_discovery/schedules/_find`

## NOTES

The feature is hidden behind the feature flag:

> xpack.securitySolution.enableExperimental: ['assistantAttackDiscoverySchedulingEnabled']

## cURL examples

<details>
  <summary>Create AD scheduling rule route</summary>

```curl
curl --location 'http://localhost:5601/internal/elastic_assistant/attack_discovery/schedules' \
--header 'kbn-xsrf: true' \
--header 'elastic-api-version: 1' \
--header 'x-elastic-internal-origin: security-solution' \
--header 'Content-Type: application/json' \
--data '{
    "name": "Test Schedule",
    "schedule": {
        "interval": "10m"
    },
    "params": {
        "alertsIndexPattern": ".alerts-security.alerts-default",
        "apiConfig": {
            "connectorId": "gpt-4o",
            "actionTypeId": ".gen-ai"
        },
        "end": "now",
        "size": 100,
        "start": "now-24h"
    }
}'
```

</details>

<details>
  <summary>Read/Get AD scheduling rule by id route</summary>

```curl
curl --location 'http://localhost:5601/internal/elastic_assistant/attack_discovery/schedules/{id}' \
--header 'kbn-xsrf: true' \
--header 'elastic-api-version: 1' \
--header 'x-elastic-internal-origin: security-solution'
```

</details>

<details>
  <summary>Update AD scheduling rule by id route</summary>

```curl
curl --location --request PUT 'http://localhost:5601/internal/elastic_assistant/attack_discovery/schedules/{id}' \
--header 'kbn-xsrf: true' \
--header 'elastic-api-version: 1' \
--header 'x-elastic-internal-origin: security-solution' \
--header 'Content-Type: application/json' \
--data '{
    "name": "Test Schedule - Updated",
    "schedule": {
        "interval": "123m"
    },
    "params": {
        "alertsIndexPattern": ".alerts-security.alerts-default",
        "apiConfig": {
            "connectorId": "gpt-4o",
            "actionTypeId": ".gen-ai"
        },
        "end": "now",
        "size": 35,
        "start": "now-24h"
    },
    "actions": []
}'
```

</details>

<details>
  <summary>Delete AD scheduling rule by id route</summary>

```curl
curl --location --request DELETE 'http://localhost:5601/internal/elastic_assistant/attack_discovery/schedules/{id}' \
--header 'kbn-xsrf: true' \
--header 'elastic-api-version: 1' \
--header 'x-elastic-internal-origin: security-solution'
```

</details>

<details>
  <summary>Enable AD scheduling rule by id route</summary>

```curl
curl --location --request POST 'http://localhost:5601/internal/elastic_assistant/attack_discovery/schedules/{id}/_enable' \
--header 'kbn-xsrf: true' \
--header 'elastic-api-version: 1' \
--header 'x-elastic-internal-origin: security-solution'
```

</details>

<details>
  <summary>Disable AD scheduling rule by id route</summary>

```curl
curl --location --request POST 'http://localhost:5601/internal/elastic_assistant/attack_discovery/schedules/{id}/_disable' \
--header 'kbn-xsrf: true' \
--header 'elastic-api-version: 1' \
--header 'x-elastic-internal-origin: security-solution'
```

</details>

<details>
  <summary>Find all existing AD scheduling rules route</summary>

```curl
curl --location 'http://localhost:5601/internal/elastic_assistant/attack_discovery/schedules/_find' \
--header 'kbn-xsrf: true' \
--header 'elastic-api-version: 1' \
--header 'x-elastic-internal-origin: security-solution'
```

</details>


<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

e40pud committed:
[Attack Discovery][Scheduling] Attack Discovery scheduling rule management (#12003)

e40pud commented on x-pack/solutions/security/packages/features/src/attack_discovery/kibana_features.ts:
```diff
@@ -35,6 +43,10 @@ export const getAttackDiscoveryBaseKibanaFeature = (): BaseKibanaFeatureConfig =
         all: [],
         read: [],
       },
+      alerting: {
```
Same alerts privileges as we have for the detection alerts https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/packages/features/src/security/v2_features/kibana_features.ts#L95

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

e40pud committed:
Resolve integration tests

e40pud committed:
Fix integration tests

e40pud commented on x-pack/test/api_integration/apis/maps/maps_telemetry.ts:
```diff
@@ -36,8 +36,8 @@ export default function ({ getService }: FtrProviderContext) {
-      expect(geoPointFieldStats.count).to.be(55);
-      expect(geoPointFieldStats.index_count).to.be(12);
+      expect(geoPointFieldStats.count).to.be(63);
+      expect(geoPointFieldStats.index_count).to.be(13);
```
Updated because we added a new attack discovery alerts aad index that maps the ecs fields. Similarly to this https://github.com/elastic/kibana/pull/194322/files#r1781620194 and this https://github.com/elastic/kibana/pull/184541/files#r1628183656.

nreese reviewed (approved):
kibana-presentation integration test assertion update changes LGTM
code review only

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/289216)
* Commit: f3bb8bef174d10cc96049a22df0e4cddd20a617c

### Failed CI Steps
* [Jest Tests #16](https://buildkite.com/elastic/kibana-pull-request/builds/289216#0195f2c0-05cf-49ee-8fe1-b47125178bdd)
* [FTR Configs #117](https://buildkite.com/elastic/kibana-pull-request/builds/289216#0195f2c0-0659-415c-87c6-79ba149ddd82)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/289216#0195f2c0-05cf-49ee-8fe1-b47125178bdd) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/289216/jobs/0195f2c0-05cf-49ee-8fe1-b47125178bdd/artifacts/0195f2ed-bd42-497a-8e50-8d514e69ab54) Jest Tests #<span></span>16 / Event filter form Details and Conditions should change description

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/fcc7244da2b801ce5e59017a00e3ce19a20b01ce) | [after](https://github.com/elastic/kibana/commit/f3bb8bef174d10cc96049a22df0e4cddd20a617c) | diff |
| --- | --- | --- | --- |
| `automaticImport` | 726 | 729 | +3 |
| `securitySolution` | 7170 | 7173 | +3 |
| total |  |  | +6 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/fcc7244da2b801ce5e59017a00e3ce19a20b01ce) | [after](https://github.com/elastic/kibana/commit/f3bb8bef174d10cc96049a22df0e4cddd20a617c) | diff |
| --- | --- | --- | --- |
| `@kbn/elastic-assistant-common` | 437 | 493 | +56 |
| `elasticAssistant` | 44 | 46 | +2 |
| total |  |  | +58 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/fcc7244da2b801ce5e59017a00e3ce19a20b01ce) | [after](https://github.com/elastic/kibana/commit/f3bb8bef174d10cc96049a22df0e4cddd20a617c) | diff |
| --- | --- | --- | --- |
| `@kbn/elastic-assistant-common` | 510 | 578 | +68 |
| `elasticAssistant` | 60 | 62 | +2 |
| total |  |  | +70 |


</details>


### History
* :broken_heart: [Build #289124](https://buildkite.com/elastic/kibana-pull-request/builds/289124) failed e2604301d25bedb4324d1431fda9988acf87d362
* :broken_heart: [Build #289076](https://buildkite.com/elastic/kibana-pull-request/builds/289076) failed b8950611be0c7fac3a0166f419071ea41451f96b

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289216","number":289216,"commit":"f3bb8bef174d10cc96049a22df0e4cddd20a617c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289124","number":289124,"commit":"e2604301d25bedb4324d1431fda9988acf87d362"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289076","number":289076,"commit":"b8950611be0c7fac3a0166f419071ea41451f96b"}],"number":289216}
buildkite-pr-comment-->

e40pud committed:
Last Execution

andrew-goldstein commented on x-pack/solutions/security/plugins/elastic_assistant/server/routes/attack_discovery/schedules/enable.ts:
```diff
@@ -0,0 +1,98 @@
+  router: IRouter<ElasticAssistantRequestHandlerContext>
+): void => {
+  router.versioned
+    .put({
```
Consider `put` -> `post`, because this route has side effects

e40pud replied:
done in https://github.com/elastic/kibana/pull/216656/commits/d167ab0c749161c3c9e22fd949d1af479da97bc0

andrew-goldstein commented on x-pack/solutions/security/plugins/elastic_assistant/server/routes/attack_discovery/schedules/disable.ts:
```diff
@@ -0,0 +1,98 @@
+  router: IRouter<ElasticAssistantRequestHandlerContext>
+): void => {
+  router.versioned
+    .put({
```
Consider `put` -> `post`, because this route has side effects

e40pud replied:
done in https://github.com/elastic/kibana/pull/216656/commits/d167ab0c749161c3c9e22fd949d1af479da97bc0

andrew-goldstein commented on x-pack/platform/packages/shared/kbn-elastic-assistant-common/impl/schemas/attack_discovery/crud_attack_discovery_schedules_route.schema.yaml:
```diff
@@ -0,0 +1,218 @@
+      x-labels: [ess, serverless]
+      operationId: GetAttackDiscoverySchedules
+      description: Gets attack discovery schedule
+      summary: Gets attack discovery schedule via the Elastic Assistant
```
> Gets attack discovery schedule via the Elastic Assistant

nit: I'm wondering if _via the Elastic Assistant_ is applicable in this summary (and the others in this route)

e40pud replied:
done in https://github.com/elastic/kibana/pull/216656/commits/89adba597b631e0c7d4a055927a7fb2543cc61b7

andrew-goldstein commented on x-pack/platform/packages/shared/kbn-elastic-assistant-common/impl/schemas/attack_discovery/crud_attack_discovery_schedules_route.schema.yaml:
```diff
@@ -0,0 +1,218 @@
+      x-codegen-enabled: true
+      x-labels: [ess, serverless]
+      operationId: UpdateAttackDiscoverySchedules
+      description: Updates attack discovery schedule
```
Consider adding a note that the `_enable` and `_disable` routes should be used (instead of this `put`) to modify the enabled state of the schedule.

andrew-goldstein commented on x-pack/solutions/security/plugins/elastic_assistant/server/routes/attack_discovery/schedules/find.test.ts:
```diff
@@ -0,0 +1,133 @@
+    });
+  });
+
+  it('should handle `dataClient.createSchedule` error', async () => {
```
nit: Consider `createSchedule` -> `findSchedules`

e40pud replied:
done in https://github.com/elastic/kibana/pull/216656/commits/89adba597b631e0c7d4a055927a7fb2543cc61b7

andrew-goldstein commented on x-pack/solutions/security/plugins/elastic_assistant/server/routes/attack_discovery/schedules/update.test.ts:
```diff
@@ -0,0 +1,126 @@
+    });
+  });
+
+  it('should handle `dataClient.createSchedule` error', async () => {
```
nit:  Consider `createSchedule` -> `updateSchedule`

e40pud replied:
done in https://github.com/elastic/kibana/pull/216656/commits/89adba597b631e0c7d4a055927a7fb2543cc61b7

andrew-goldstein reviewed (approved):
Thanks @e40pud for the new scheduling APIs!
✅ Desk tested locally
LGTM 🚀

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/12003-rules-mgmt

e40pud commented on x-pack/solutions/security/plugins/elastic_assistant/server/plugin.ts:
```diff
@@ -89,6 +90,13 @@ export class ElasticAssistantPlugin
     registerRoutes(router, this.logger, this.getElserId);
 
+    // Register Attack Discovery Schedule type
+    plugins.alerting.registerType(
```
Added feature flags related changes in https://github.com/elastic/kibana/pull/217306

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/290642)
* Commit: c738af444ab5f266f069e4546b100a3997c0d89e

### Failed CI Steps
* [Jest Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/290642#01960fc3-9e29-41dd-8bd2-47aa3406ff6b)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/290642#01960fc3-9e29-41dd-8bd2-47aa3406ff6b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/290642/jobs/01960fc3-9e29-41dd-8bd2-47aa3406ff6b/artifacts/01960fea-5c79-4231-8ba5-4be1b82d7cac) Jest Tests #<span></span>10 / <TableTab /> should filter the table correctly

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/135b57bbaca9e77f091002dcea2b60250b37ba3c) | [after](https://github.com/elastic/kibana/commit/c738af444ab5f266f069e4546b100a3997c0d89e) | diff |
| --- | --- | --- | --- |
| `automaticImport` | 728 | 731 | +3 |
| `securitySolution` | 7184 | 7187 | +3 |
| total |  |  | +6 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/135b57bbaca9e77f091002dcea2b60250b37ba3c) | [after](https://github.com/elastic/kibana/commit/c738af444ab5f266f069e4546b100a3997c0d89e) | diff |
| --- | --- | --- | --- |
| `@kbn/elastic-assistant-common` | 445 | 500 | +55 |
| `elasticAssistant` | 44 | 46 | +2 |
| total |  |  | +57 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/135b57bbaca9e77f091002dcea2b60250b37ba3c) | [after](https://github.com/elastic/kibana/commit/c738af444ab5f266f069e4546b100a3997c0d89e) | diff |
| --- | --- | --- | --- |
| `@kbn/elastic-assistant-common` | 518 | 585 | +67 |
| `elasticAssistant` | 60 | 62 | +2 |
| total |  |  | +69 |


</details>


### History
* :yellow_heart: [Build #289216](https://buildkite.com/elastic/kibana-pull-request/builds/289216) was flaky f3bb8bef174d10cc96049a22df0e4cddd20a617c
* :broken_heart: [Build #289124](https://buildkite.com/elastic/kibana-pull-request/builds/289124) failed e2604301d25bedb4324d1431fda9988acf87d362
* :broken_heart: [Build #289076](https://buildkite.com/elastic/kibana-pull-request/builds/289076) failed b8950611be0c7fac3a0166f419071ea41451f96b

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290642","number":290642,"commit":"c738af444ab5f266f069e4546b100a3997c0d89e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289216","number":289216,"commit":"f3bb8bef174d10cc96049a22df0e4cddd20a617c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289124","number":289124,"commit":"e2604301d25bedb4324d1431fda9988acf87d362"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289076","number":289076,"commit":"b8950611be0c7fac3a0166f419071ea41451f96b"}],"number":290642}
buildkite-pr-comment-->

e40pud committed:
Switch `enable` and `disable` routes to use `POST` method

e40pud committed:
Review feedback: fix namings

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/290741)
* Commit: 89adba597b631e0c7d4a055927a7fb2543cc61b7

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/135b57bbaca9e77f091002dcea2b60250b37ba3c) | [after](https://github.com/elastic/kibana/commit/89adba597b631e0c7d4a055927a7fb2543cc61b7) | diff |
| --- | --- | --- | --- |
| `automaticImport` | 728 | 731 | +3 |
| `securitySolution` | 7184 | 7187 | +3 |
| total |  |  | +6 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/135b57bbaca9e77f091002dcea2b60250b37ba3c) | [after](https://github.com/elastic/kibana/commit/89adba597b631e0c7d4a055927a7fb2543cc61b7) | diff |
| --- | --- | --- | --- |
| `@kbn/elastic-assistant-common` | 445 | 504 | +59 |
| `elasticAssistant` | 44 | 46 | +2 |
| total |  |  | +61 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/135b57bbaca9e77f091002dcea2b60250b37ba3c) | [after](https://github.com/elastic/kibana/commit/89adba597b631e0c7d4a055927a7fb2543cc61b7) | diff |
| --- | --- | --- | --- |
| `@kbn/elastic-assistant-common` | 518 | 591 | +73 |
| `elasticAssistant` | 60 | 62 | +2 |
| total |  |  | +75 |


</details>


### History
* :yellow_heart: [Build #290642](https://buildkite.com/elastic/kibana-pull-request/builds/290642) was flaky c738af444ab5f266f069e4546b100a3997c0d89e
* :yellow_heart: [Build #289216](https://buildkite.com/elastic/kibana-pull-request/builds/289216) was flaky f3bb8bef174d10cc96049a22df0e4cddd20a617c
* :broken_heart: [Build #289124](https://buildkite.com/elastic/kibana-pull-request/builds/289124) failed e2604301d25bedb4324d1431fda9988acf87d362
* :broken_heart: [Build #289076](https://buildkite.com/elastic/kibana-pull-request/builds/289076) failed b8950611be0c7fac3a0166f419071ea41451f96b

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290741","number":290741,"commit":"89adba597b631e0c7d4a055927a7fb2543cc61b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290642","number":290642,"commit":"c738af444ab5f266f069e4546b100a3997c0d89e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289216","number":289216,"commit":"f3bb8bef174d10cc96049a22df0e4cddd20a617c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289124","number":289124,"commit":"e2604301d25bedb4324d1431fda9988acf87d362"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289076","number":289076,"commit":"b8950611be0c7fac3a0166f419071ea41451f96b"}],"number":290741}
buildkite-pr-comment-->

pmuellr commented on src/platform/packages/shared/kbn-alerts-as-data-utils/src/schemas/generated/security_attackdiscovery_schema.ts:
```diff
@@ -0,0 +1,145 @@
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+// ---------------------------------- WARNING ----------------------------------
+// this file was generated, and should not be edited by hand
```
Usually these files are generated from files with ES mappings, I believe mainly in the peer directory `src/field_maps`. Not seeing where it might have been generated for.

Which makes me wonder if we've defined mappings for these fields in the alerts documents.  Perhaps this is done differently for security rules, or this is part of a feature branch where some of that work was already done.  

In any case, we should make sure we have the ES mappings in place for these field, if we want them indexed.

e40pud replied:
The mappings are defined in this file https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/schedules/fields/field_map.ts.

I followed other solutions including security while adding these mappings and this new alert type. Similarly to o11y, ml, security, stack etc. the mappings are defined inside those solutions and the generated schemas then added via FTR test script inside the `src/platform/packages/shared/kbn-alerts-as-data-utils/src/schemas/generated/`.

Here is example of similarly defined mappings https://github.com/elastic/kibana/blob/main/x-pack/solutions/observability/plugins/apm/server/routes/alerts/register_apm_rule_types.ts

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/12003-rules-mgmt

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291063)
* Commit: 71eb297cec25a5ac137a824662a5a72c98b2e8ea

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/291063#01961484-64be-4bf4-ba31-a1656284f541)

### History
* :green_heart: [Build #290741](https://buildkite.com/elastic/kibana-pull-request/builds/290741) succeeded 89adba597b631e0c7d4a055927a7fb2543cc61b7
* :yellow_heart: [Build #290642](https://buildkite.com/elastic/kibana-pull-request/builds/290642) was flaky c738af444ab5f266f069e4546b100a3997c0d89e
* :yellow_heart: [Build #289216](https://buildkite.com/elastic/kibana-pull-request/builds/289216) was flaky f3bb8bef174d10cc96049a22df0e4cddd20a617c
* :broken_heart: [Build #289124](https://buildkite.com/elastic/kibana-pull-request/builds/289124) failed e2604301d25bedb4324d1431fda9988acf87d362
* :broken_heart: [Build #289076](https://buildkite.com/elastic/kibana-pull-request/builds/289076) failed b8950611be0c7fac3a0166f419071ea41451f96b

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291063","number":291063,"commit":"71eb297cec25a5ac137a824662a5a72c98b2e8ea"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290741","number":290741,"commit":"89adba597b631e0c7d4a055927a7fb2543cc61b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290642","number":290642,"commit":"c738af444ab5f266f069e4546b100a3997c0d89e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289216","number":289216,"commit":"f3bb8bef174d10cc96049a22df0e4cddd20a617c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289124","number":289124,"commit":"e2604301d25bedb4324d1431fda9988acf87d362"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289076","number":289076,"commit":"b8950611be0c7fac3a0166f419071ea41451f96b"}],"number":291063}
buildkite-pr-comment-->

e40pud committed:
Switch to using `core.featureFlags` service

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291097)
* Commit: d4285c4b8a10a49818e1157fe4ecbd7fbf4af859

### Failed CI Steps
* [Linting (with types)](https://buildkite.com/elastic/kibana-pull-request/builds/291097#019614d5-cd58-4e22-bb98-c7d33bf84878)

### History
* :broken_heart: [Build #291063](https://buildkite.com/elastic/kibana-pull-request/builds/291063) failed 71eb297cec25a5ac137a824662a5a72c98b2e8ea
* :green_heart: [Build #290741](https://buildkite.com/elastic/kibana-pull-request/builds/290741) succeeded 89adba597b631e0c7d4a055927a7fb2543cc61b7
* :yellow_heart: [Build #290642](https://buildkite.com/elastic/kibana-pull-request/builds/290642) was flaky c738af444ab5f266f069e4546b100a3997c0d89e
* :yellow_heart: [Build #289216](https://buildkite.com/elastic/kibana-pull-request/builds/289216) was flaky f3bb8bef174d10cc96049a22df0e4cddd20a617c
* :broken_heart: [Build #289124](https://buildkite.com/elastic/kibana-pull-request/builds/289124) failed e2604301d25bedb4324d1431fda9988acf87d362
* :broken_heart: [Build #289076](https://buildkite.com/elastic/kibana-pull-request/builds/289076) failed b8950611be0c7fac3a0166f419071ea41451f96b

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291097","number":291097,"commit":"d4285c4b8a10a49818e1157fe4ecbd7fbf4af859"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291063","number":291063,"commit":"71eb297cec25a5ac137a824662a5a72c98b2e8ea"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290741","number":290741,"commit":"89adba597b631e0c7d4a055927a7fb2543cc61b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290642","number":290642,"commit":"c738af444ab5f266f069e4546b100a3997c0d89e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289216","number":289216,"commit":"f3bb8bef174d10cc96049a22df0e4cddd20a617c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289124","number":289124,"commit":"e2604301d25bedb4324d1431fda9988acf87d362"}],"number":291097}
buildkite-pr-comment-->

e40pud committed:
Fix linting

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291119)
* Commit: 61ab6566abe40cafd9f2b4bb0c46f98e64709a07

### Failed CI Steps
* [Jest Integration Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/291119#0196151a-c314-4aee-ae04-db61517d029b)
* [FTR Configs #33](https://buildkite.com/elastic/kibana-pull-request/builds/291119#0196151a-c33b-409a-8ae1-b8c04f7e933e)
* [FTR Configs #33](https://buildkite.com/elastic/kibana-pull-request/builds/291119#0196153e-76a8-4fe2-9377-25a848e5ef04)
* [FTR Configs #83](https://buildkite.com/elastic/kibana-pull-request/builds/291119#0196151a-c36b-48bb-92f3-f5e0370e7870)
* [FTR Configs #86](https://buildkite.com/elastic/kibana-pull-request/builds/291119#0196151a-c36e-4995-a02f-23b39ef85298)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291119#0196151a-c314-4aee-ae04-db61517d029b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291119/jobs/0196151a-c314-4aee-ae04-db61517d029b/artifacts/01961542-7886-4607-8bfc-1e47de269165) Jest Integration Tests #<span></span>4 / Alert as data fields checks detect AAD fields changes for: attack-discovery
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291119#0196151a-c314-4aee-ae04-db61517d029b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291119/jobs/0196151a-c314-4aee-ae04-db61517d029b/artifacts/01961542-7886-4e0c-8e28-2c67683f179b) Jest Integration Tests #<span></span>4 / Alert as data fields checks ensure rule types list up to date
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291119#0196151a-c36b-48bb-92f3-f5e0370e7870) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291119/jobs/0196151a-c36b-48bb-92f3-f5e0370e7870/artifacts/01961540-d5f1-409b-a78c-52130dc18b72) FTR Configs #<span></span>83 / Alerting check registered rule types should list all registered rule types
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291119#0196151a-c33b-409a-8ae1-b8c04f7e933e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291119/jobs/0196151a-c33b-409a-8ae1-b8c04f7e933e/artifacts/0196153e-6347-4256-bc14-04d23a59c23e) FTR Configs #<span></span>33 / Maps endpoints apis maps_telemetry should return the correct telemetry values for map saved objects
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291119#0196153e-76a8-4fe2-9377-25a848e5ef04) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291119/jobs/0196153e-76a8-4fe2-9377-25a848e5ef04/artifacts/01961544-2edf-4f83-98ce-81804781bf62) FTR Configs #<span></span>33 / Maps endpoints apis maps_telemetry should return the correct telemetry values for map saved objects
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291119#0196151a-c36e-4995-a02f-23b39ef85298) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291119/jobs/0196151a-c36e-4995-a02f-23b39ef85298/artifacts/01961540-50bc-477e-baf9-8b51cadac070) FTR Configs #<span></span>86 / task_manager check_registered_task_types should check changes on all registered task types

### History
* :broken_heart: [Build #291097](https://buildkite.com/elastic/kibana-pull-request/builds/291097) failed d4285c4b8a10a49818e1157fe4ecbd7fbf4af859
* :broken_heart: [Build #291063](https://buildkite.com/elastic/kibana-pull-request/builds/291063) failed 71eb297cec25a5ac137a824662a5a72c98b2e8ea
* :green_heart: [Build #290741](https://buildkite.com/elastic/kibana-pull-request/builds/290741) succeeded 89adba597b631e0c7d4a055927a7fb2543cc61b7
* :yellow_heart: [Build #290642](https://buildkite.com/elastic/kibana-pull-request/builds/290642) was flaky c738af444ab5f266f069e4546b100a3997c0d89e
* :yellow_heart: [Build #289216](https://buildkite.com/elastic/kibana-pull-request/builds/289216) was flaky f3bb8bef174d10cc96049a22df0e4cddd20a617c

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291119","number":291119,"commit":"61ab6566abe40cafd9f2b4bb0c46f98e64709a07"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291097","number":291097,"commit":"d4285c4b8a10a49818e1157fe4ecbd7fbf4af859"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291063","number":291063,"commit":"71eb297cec25a5ac137a824662a5a72c98b2e8ea"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290741","number":290741,"commit":"89adba597b631e0c7d4a055927a7fb2543cc61b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290642","number":290642,"commit":"c738af444ab5f266f069e4546b100a3997c0d89e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289216","number":289216,"commit":"f3bb8bef174d10cc96049a22df0e4cddd20a617c"}],"number":291119}
buildkite-pr-comment-->

e40pud committed:
Better alert naming: `SecurityAttackDiscoveryAlertRequired` instead of `SecurityAttackdiscoveryAlertRequired`

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291178)
* Commit: 36f2c44d3c60ea84146fc2f7131d0ad8cfbdea7d

### Failed CI Steps
* [Jest Integration Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/291178#0196159e-6a24-4238-9e65-59c0e7e5fb72)
* [FTR Configs #63](https://buildkite.com/elastic/kibana-pull-request/builds/291178#0196159e-6a54-4887-a116-17385649067d)
* [FTR Configs #82](https://buildkite.com/elastic/kibana-pull-request/builds/291178#0196159e-6a5f-4476-ad4c-16b3c0e91d4b)
* [FTR Configs #89](https://buildkite.com/elastic/kibana-pull-request/builds/291178#0196159e-6a64-4116-8de0-6e47f21b24d5)
* [FTR Configs #89](https://buildkite.com/elastic/kibana-pull-request/builds/291178#019615c3-3f72-44ef-a862-efe1a87c2344)
* [FTR Configs #119](https://buildkite.com/elastic/kibana-pull-request/builds/291178#0196159e-6a74-4a9d-ad55-d6a2d45443ba)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291178#0196159e-6a24-4238-9e65-59c0e7e5fb72) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291178/jobs/0196159e-6a24-4238-9e65-59c0e7e5fb72/artifacts/019615cc-5bf9-41a5-a2af-123d1c23cceb) Jest Integration Tests #<span></span>1 / Alert as data fields checks detect AAD fields changes for: attack-discovery
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291178#0196159e-6a24-4238-9e65-59c0e7e5fb72) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291178/jobs/0196159e-6a24-4238-9e65-59c0e7e5fb72/artifacts/019615cc-5bf8-49ee-8c8f-84bdff2b4f19) Jest Integration Tests #<span></span>1 / Alert as data fields checks ensure rule types list up to date
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291178#0196159e-6a5f-4476-ad4c-16b3c0e91d4b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291178/jobs/0196159e-6a5f-4476-ad4c-16b3c0e91d4b/artifacts/019615c3-76f5-44af-a4d6-316b090bfc02) FTR Configs #<span></span>82 / Alerting check registered rule types should list all registered rule types
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291178#0196159e-6a74-4a9d-ad55-d6a2d45443ba) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291178/jobs/0196159e-6a74-4a9d-ad55-d6a2d45443ba/artifacts/019615ca-0542-47f6-a5cb-2cb2faa63a0b) FTR Configs #<span></span>119 / Maps endpoints apis maps_telemetry should return the correct telemetry values for map saved objects
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291178#0196159e-6a64-4116-8de0-6e47f21b24d5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291178/jobs/0196159e-6a64-4116-8de0-6e47f21b24d5/artifacts/019615c3-2c2c-41b5-b182-f3afa02a6a13) FTR Configs #<span></span>89 / task_manager check_registered_task_types should check changes on all registered task types
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291178#019615c3-3f72-44ef-a862-efe1a87c2344) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291178/jobs/019615c3-3f72-44ef-a862-efe1a87c2344/artifacts/019615cc-f934-4eea-9374-eccf4b54839f) FTR Configs #<span></span>89 / task_manager check_registered_task_types should check changes on all registered task types

### History
* :broken_heart: [Build #291119](https://buildkite.com/elastic/kibana-pull-request/builds/291119) failed 61ab6566abe40cafd9f2b4bb0c46f98e64709a07
* :broken_heart: [Build #291097](https://buildkite.com/elastic/kibana-pull-request/builds/291097) failed d4285c4b8a10a49818e1157fe4ecbd7fbf4af859
* :broken_heart: [Build #291063](https://buildkite.com/elastic/kibana-pull-request/builds/291063) failed 71eb297cec25a5ac137a824662a5a72c98b2e8ea
* :green_heart: [Build #290741](https://buildkite.com/elastic/kibana-pull-request/builds/290741) succeeded 89adba597b631e0c7d4a055927a7fb2543cc61b7
* :yellow_heart: [Build #290642](https://buildkite.com/elastic/kibana-pull-request/builds/290642) was flaky c738af444ab5f266f069e4546b100a3997c0d89e

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291178","number":291178,"commit":"36f2c44d3c60ea84146fc2f7131d0ad8cfbdea7d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291119","number":291119,"commit":"61ab6566abe40cafd9f2b4bb0c46f98e64709a07"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291097","number":291097,"commit":"d4285c4b8a10a49818e1157fe4ecbd7fbf4af859"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291063","number":291063,"commit":"71eb297cec25a5ac137a824662a5a72c98b2e8ea"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290741","number":290741,"commit":"89adba597b631e0c7d4a055927a7fb2543cc61b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290642","number":290642,"commit":"c738af444ab5f266f069e4546b100a3997c0d89e"}],"number":291178}
buildkite-pr-comment-->

e40pud committed:
Fix tests

Became broken since we register new alerting rule conditionally based on feature flag (which is off by default in tests).

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/12003-rules-mgmt

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291292)
* Commit: 9ddd16a6850388ad4dd5ea0caabb79d87d34bce6

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/fb961ec880ca069dd77cc83b9369e5351d787e5f) | [after](https://github.com/elastic/kibana/commit/9ddd16a6850388ad4dd5ea0caabb79d87d34bce6) | diff |
| --- | --- | --- | --- |
| `automaticImport` | 728 | 731 | +3 |
| `securitySolution` | 7191 | 7194 | +3 |
| total |  |  | +6 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/fb961ec880ca069dd77cc83b9369e5351d787e5f) | [after](https://github.com/elastic/kibana/commit/9ddd16a6850388ad4dd5ea0caabb79d87d34bce6) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-as-data-utils` | 33 | 34 | +1 |
| `@kbn/elastic-assistant-common` | 446 | 505 | +59 |
| `elasticAssistant` | 44 | 46 | +2 |
| total |  |  | +62 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/fb961ec880ca069dd77cc83b9369e5351d787e5f) | [after](https://github.com/elastic/kibana/commit/9ddd16a6850388ad4dd5ea0caabb79d87d34bce6) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-as-data-utils` | 33 | 34 | +1 |
| `@kbn/elastic-assistant-common` | 519 | 592 | +73 |
| `elasticAssistant` | 60 | 62 | +2 |
| total |  |  | +76 |


</details>


### History
* :broken_heart: [Build #291218](https://buildkite.com/elastic/kibana-pull-request/builds/291218) failed 9312d2127df68567195a04c13b8e7a7925a7f2c2
* :broken_heart: [Build #291178](https://buildkite.com/elastic/kibana-pull-request/builds/291178) failed 36f2c44d3c60ea84146fc2f7131d0ad8cfbdea7d
* :broken_heart: [Build #291119](https://buildkite.com/elastic/kibana-pull-request/builds/291119) failed 61ab6566abe40cafd9f2b4bb0c46f98e64709a07
* :broken_heart: [Build #291097](https://buildkite.com/elastic/kibana-pull-request/builds/291097) failed d4285c4b8a10a49818e1157fe4ecbd7fbf4af859
* :broken_heart: [Build #291063](https://buildkite.com/elastic/kibana-pull-request/builds/291063) failed 71eb297cec25a5ac137a824662a5a72c98b2e8ea

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291292","number":291292,"commit":"9ddd16a6850388ad4dd5ea0caabb79d87d34bce6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291218","number":291218,"commit":"9312d2127df68567195a04c13b8e7a7925a7f2c2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291178","number":291178,"commit":"36f2c44d3c60ea84146fc2f7131d0ad8cfbdea7d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291119","number":291119,"commit":"61ab6566abe40cafd9f2b4bb0c46f98e64709a07"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291097","number":291097,"commit":"d4285c4b8a10a49818e1157fe4ecbd7fbf4af859"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291063","number":291063,"commit":"71eb297cec25a5ac137a824662a5a72c98b2e8ea"}],"number":291292}
buildkite-pr-comment-->

pmuellr commented:
I'm curious why the a rule is used here instead of a taskmanager task.  Task manager tasks are generally more open-ended, but there's also a lot of built-in stuff with rules like metrics, APM, logging, etc.  Not a problem, just curious.

@mikecote Code looks good to me, but I noticed you added it to our board last week.  The rule executor is empty at the moment, so presumably that'll come in the future :-).  Was wondering if there was anything specific to look for here ...

mikecote commented:
>@mikecote Code looks good to me, but I noticed you added it to our board last week. The rule executor is empty at the moment, so presumably that'll come in the future :-). Was wondering if there was anything specific to look for here ...

@pmuellr I think the PR scope might have changed since I added it. But I basically added it for a codeowners review. We should see a follow up PR soon for the rule type @e40pud ?

e40pud commented:
> @pmuellr I think the PR scope might have changed since I added it. But I basically added it for a codeowners review. We should see a follow up PR soon for the rule type @e40pud ?

@mikecote correct. There will be a follow up PR and I will add you both to it to review. With this PR I setup rules management APIs and introduced a new rule type.

@pmuellr we decided to move with the alerting framework rules because of:
- After many discussions everyone aligned on a vision of attack discoveries as alerts/signals. At some point they will be presented on the same page with existing detection alerts/signals in security solution.
- Time limits that we have. This feature needs to be mostly done by the end of April and fully available in 8.19/9.1 versions.
- We need to have notifications/actions support (including conditional actions) and we did not want to reimplement all this due to time limits. Same applies to other existing tools handled already by the framework.

semd commented on x-pack/solutions/security/packages/features/src/attack_discovery/kibana_features.ts:
```diff
@@ -9,9 +9,16 @@ import { DEFAULT_APP_CATEGORIES } from '@kbn/core-application-common';
 
+const alertingFeatures = [
+  {
+    ruleTypeId: 'attack-discovery',
```
NIT: consider using a constant for this `ruleTypeId`

e40pud replied:
It actually exists, just forgot to use it here. Thanks for catching this one!

e40pud replied:
done in https://github.com/elastic/kibana/pull/216656/commits/bfb60dbc15491f1dd6fd7c28ee4eeeadd55e360f

semd reviewed (approved):
LGTM

e40pud committed:
Review feedback: use constants

e40pud committed:
Merge branch 'main' into security/genai/ad-schedule/12003-rules-mgmt

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291659)
* Commit: 1962ca4bdc9e9ffbff237ad52f3f5604d7ad754d

### Failed CI Steps
* [FTR Configs #49](https://buildkite.com/elastic/kibana-pull-request/builds/291659#01961b76-f661-40ef-b42c-c63f3547bfba)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/291659#01961b76-f661-40ef-b42c-c63f3547bfba) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/291659/jobs/01961b76-f661-40ef-b42c-c63f3547bfba/artifacts/01961b9a-48fc-41a9-9b8d-9a0185c0c46f) FTR Configs #<span></span>49 / discover/group1 discover accessibility top nav menu buttons should return focus to the alerts button when dismissing the alerts popover

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/b6907a54d8f9ae23bc58be90dc33dcfa24615569) | [after](https://github.com/elastic/kibana/commit/1962ca4bdc9e9ffbff237ad52f3f5604d7ad754d) | diff |
| --- | --- | --- | --- |
| `automaticImport` | 728 | 731 | +3 |
| `securitySolution` | 7192 | 7195 | +3 |
| total |  |  | +6 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/b6907a54d8f9ae23bc58be90dc33dcfa24615569) | [after](https://github.com/elastic/kibana/commit/1962ca4bdc9e9ffbff237ad52f3f5604d7ad754d) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-as-data-utils` | 33 | 34 | +1 |
| `@kbn/elastic-assistant-common` | 446 | 505 | +59 |
| `elasticAssistant` | 44 | 46 | +2 |
| total |  |  | +62 |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/b6907a54d8f9ae23bc58be90dc33dcfa24615569) | [after](https://github.com/elastic/kibana/commit/1962ca4bdc9e9ffbff237ad52f3f5604d7ad754d) | diff |
| --- | --- | --- | --- |
| `@kbn/alerts-as-data-utils` | 33 | 34 | +1 |
| `@kbn/elastic-assistant-common` | 519 | 592 | +73 |
| `elasticAssistant` | 60 | 62 | +2 |
| total |  |  | +76 |


</details>


### History
* :green_heart: [Build #291292](https://buildkite.com/elastic/kibana-pull-request/builds/291292) succeeded 9ddd16a6850388ad4dd5ea0caabb79d87d34bce6
* :broken_heart: [Build #291218](https://buildkite.com/elastic/kibana-pull-request/builds/291218) failed 9312d2127df68567195a04c13b8e7a7925a7f2c2
* :broken_heart: [Build #291178](https://buildkite.com/elastic/kibana-pull-request/builds/291178) failed 36f2c44d3c60ea84146fc2f7131d0ad8cfbdea7d
* :broken_heart: [Build #291119](https://buildkite.com/elastic/kibana-pull-request/builds/291119) failed 61ab6566abe40cafd9f2b4bb0c46f98e64709a07
* :broken_heart: [Build #291097](https://buildkite.com/elastic/kibana-pull-request/builds/291097) failed d4285c4b8a10a49818e1157fe4ecbd7fbf4af859
* :broken_heart: [Build #291063](https://buildkite.com/elastic/kibana-pull-request/builds/291063) failed 71eb297cec25a5ac137a824662a5a72c98b2e8ea

cc @e40pud

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291659","number":291659,"commit":"1962ca4bdc9e9ffbff237ad52f3f5604d7ad754d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291292","number":291292,"commit":"9ddd16a6850388ad4dd5ea0caabb79d87d34bce6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291218","number":291218,"commit":"9312d2127df68567195a04c13b8e7a7925a7f2c2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291178","number":291178,"commit":"36f2c44d3c60ea84146fc2f7131d0ad8cfbdea7d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291119","number":291119,"commit":"61ab6566abe40cafd9f2b4bb0c46f98e64709a07"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291097","number":291097,"commit":"d4285c4b8a10a49818e1157fe4ecbd7fbf4af859"}],"number":291659}
buildkite-pr-comment-->

pmuellr commented:
@e40pud thanks for the description of task vs rule.  

I was curious to see the wrapper over alerting client in
[.../elastic_assistant/server/lib/attack_discovery/schedules/data_client/index.ts](https://github.com/elastic/kibana/pull/216656/files#diff-a71c9ccc0dc80cdbabbf47feacedee24f2ac5fad0e0cbb40c13dbf59dde62d35).   

The wrappers aren't doing much now, but in the future I imagine they will be doing some interesting things.

Folks will be able to create a rule of this type outside of this wrapper, for example using the alerting HTTP APIs.  As well as update, enable/disable, etc.  I don't think we have a way of "hiding" rule types to prevent this sort of thing.  Is that going to be ok?

We [added a capability to connectors to allow code to be run "around" some APIs](https://github.com/elastic/kibana/pull/194081) - we only added what was needed at the time.  And have talked about doing a similar sort of thing with rules.  I can't find an issue to track that (so no plan to implement now), but am curious if something like that could essentially replace your existing wrapper.

pmuellr reviewed (approved):
ResponseOps changes (minimal) LGTM

afharo reviewed (approved):
`kibana.jsonc` changes LGTM

e40pud commented:
> @e40pud thanks for the description of task vs rule.
> 
> I was curious to see the wrapper over alerting client in [.../elastic_assistant/server/lib/attack_discovery/schedules/data_client/index.ts](https://github.com/elastic/kibana/pull/216656/files#diff-a71c9ccc0dc80cdbabbf47feacedee24f2ac5fad0e0cbb40c13dbf59dde62d35).
> 
> The wrappers aren't doing much now, but in the future I imagine they will be doing some interesting things.
> 
> Folks will be able to create a rule of this type outside of this wrapper, for example using the alerting HTTP APIs. As well as update, enable/disable, etc. I don't think we have a way of "hiding" rule types to prevent this sort of thing. Is that going to be ok?
> 
> We [added a capability to connectors to allow code to be run "around" some APIs](https://github.com/elastic/kibana/pull/194081) - we only added what was needed at the time. And have talked about doing a similar sort of thing with rules. I can't find an issue to track that (so no plan to implement now), but am curious if something like that could essentially replace your existing wrapper.

This wrapper is mostly to add some common utils in those wrappers and to align API differences:
* passed parameters
* conversions alerting framework <=> security solution. We will strip out some of the fields returned by alerting framework and return those as attack discovery schedule type
* potentially it will be one point of logging/telemetry reporting (if ever needed)

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/14376549179

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.x|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 216656
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

e40pud commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/217794">](https://github.com/elastic/kibana/pull/217794)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
